polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -146,6 +146,14 @@ ValueCategory MLIRScanner::VisitParenExpr(clang::ParenExpr *Expr) {
   return Visit(Expr->getSubExpr());
 }
 
+template <typename T>
+mlir::MemRefType MLIRScanner::SYCLGetElementType(T SYCLType, unsigned ElemIndex,
+                                                 llvm::ArrayRef<int64_t> Shape,
+                                                 mlir::Attribute MemorySpace) {
+  return mlir::MemRefType::get(Shape, cast<T>(SYCLType).getBody()[ElemIndex],
+                               MemRefLayoutAttrInterface(), MemorySpace);
```

Comment:
Casting from `T` to `T` here. You can drop the cast and just inline the function in the `TypeSwitch`

Reply:
I will add another type to the same case in the `TypeSwitch`, that's why created `SYCLGetElementType`. Are you ok with not inlining the function in the `TypeSwitch`?